imports/database/database-access-object.js

```diff
@@ -0,0 +1,37 @@
+export class DatabaseAccessObject {
+    constructor(collection) {
+        this._collection = collection;
+    }
+
+    find(queryObject) {
+        return this._collection.find(queryObject);
+    }
+
+    findOne(queryObject) {
+        return this._collection.findOne(queryObject);
+    }
+
+    insert(object) {
+        this._collection.schema.validate(object);
+        this._collection.insert(object);
+    }
+
+    update(id, object) {
+        let isValid = true;
+        object.keys().forEach(function (key) {
+            isValid &= this._collection.schema.newContext().validateOne(object, key);
+        });
+
+        if (!isValid) {
+            throw new Error("Update object did not pass validation");
```

Comment:
would be better if told you which keys failed (or the first one to fail) could keep a record of this in an array or if there's a way to break from forEach that would work too.